exporter/ststopologyexporter/internal/topology.go

```diff
@@ -0,0 +1,158 @@
+package internal
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+)
+
+type ComponentsCollection struct {
+	namespaces       map[string]*Component
+	services         []*Component
+	serviceInstances []*Component
+}
+
+func NewCollection() *ComponentsCollection {
+	return &ComponentsCollection{
+		make(map[string]*Component, 0),
+		make([]*Component, 0),
+		make([]*Component, 0),
+	}
+}
+
+func (c *ComponentsCollection) AddResource(attrs *pcommon.Map) bool {
+	serviceName, ok := attrs.Get("service.name")
+	if !ok {
+		return false
+	}
+	serviceNamespace, ok := attrs.Get("service.namespace")
+	if !ok {
+		return false
+	}
+	instanceId, ok := attrs.Get("service.instance.id")
+	var serviceInstanceId pcommon.Value
+	if !ok {
+		serviceInstanceId = serviceName
+	} else {
+		serviceInstanceId = instanceId
+	}
+
+	if _, ok := c.namespaces[serviceNamespace.AsString()]; !ok {
+		c.namespaces[serviceNamespace.AsString()] = &Component{
+			fmt.Sprintf("urn:opentelemetry:namespace/%s", serviceNamespace.AsString()),
+			ComponentType{
+				"namespace",
+			},
+			newData().
+				withLayer("urn:stackpack:common:layer:applications").
+				withEnvironment(attrs).
```

Comment:
Shouldn't we also then keep the `deployment.environment` in the tags and maybe some more? I've actually been asking if we could just deprecate the required environment setting in StackState...